### ISSUE TITLE
MultilineLambdaItParameter should not be triggered when parameter is not used at all

### ISSUE THREAD

**Issue by zarebski-m** [detekt/detekt#3570]
## Expected Behavior of the rule
When lambda body does not refer to implicit `it` parameter at all, the rule should not be triggered.

## Context
Spring transaction template provides an `execute` extension method where lambda parameter is `TransactionStatus`. In most cases the parameter is not used at all, e.g.:

```kotlin
tx.execute {
    val entity = entityManager.find(Entity::class.java, 1)
    // more code within transaction
    // lambda parameter is never used but the rule is triggered
}
```

**Comment by schalkms** [comment]
Did you try to use `_` in the lambda expression?

**Comment by BraisGabin** [comment]
I agree with @zarebski-m here. You could use `{ _ -> }` but that's just verbosity and it provides no real benefits to the code.
